pulser-core/pulser/channels/base_channel.py

```diff
@@ -262,6 +262,14 @@ def Local(
             min_avg_amp: The minimum average amplitude of a pulse (when not
                 zero).
         """
+        # Can't initialize a channel whose addressing is determined internally
+        for cls_field in fields(cls):
+            if cls_field.name == "addressing":
+                break
+        if not cls_field.init and not cls_field.default == MISSING:
```

Comment:
nit: The more pythonic way is
```suggestion
        if not cls_field.init and cls_field.default is not MISSING:
```